supertokens_python/recipe/thirdpartyemailpassword/email_delivery/services/backward_compatibility/__init__.py

```diff
@@ -0,0 +1,112 @@
+from supertokens_python.recipe.emailverification.interfaces import \
+    TypeEmailVerificationEmailDeliveryInput
+from supertokens_python.recipe.emailverification.types import User as EVUser
+from supertokens_python.recipe.thirdpartyemailpassword.recipeimplementation.implementation import \
+    RecipeImplementation
+from supertokens_python.recipe.thirdpartyemailpassword.types import (
+    TypeThirdPartyEmailPasswordEmailDeliveryInput, User)
+from supertokens_python.supertokens import AppInfo
+
+if TYPE_CHECKING:
+    from supertokens_python.recipe.thirdpartyemailpassword.utils import \
+        InputEmailVerificationConfig
+
+
+class BackwardCompatibilityService(EmailDeliveryInterface[TypeThirdPartyEmailPasswordEmailDeliveryInput]):
+    app_info: AppInfo
+    ev_backward_comp_service: EmailVerificationBackwardCompatibilityService
+    ep_backward_comp_service: EmailPasswordBackwardCompatibilityService
+
+    def __init__(self,
+                 app_info: AppInfo,
+                 recipe_interface_impl: RecipeImplementation,
+                 reset_password_using_token_feature: Union[InputResetPasswordUsingTokenFeature, None] = None,
+                 email_verification_feature: Union[InputEmailVerificationConfig, None] = None,
+                 ) -> None:
+        self.app_info = app_info
+        self.recipe_interface_impl = recipe_interface_impl
+
+        # Setup EmailVerification (EV) backward compatibility service
+        input_create_and_send_custom_email = email_verification_feature.create_and_send_custom_email if email_verification_feature is not None else None
+        if input_create_and_send_custom_email is None:
+            ev_create_and_send_custom_email = None
+        else:
+            async def create_and_send_custom_email(user: EVUser, link: str, user_context: Dict[str, Any]):
+                user_info = await recipe_interface_impl.get_user_by_id(user.user_id, user_context)
+                if user_info is None:
+                    raise Exception("Unknown user ID provided")
+                return await input_create_and_send_custom_email(user_info, link, user_context)
+            ev_create_and_send_custom_email = create_and_send_custom_email
+
+        self.ev_backward_comp_service = EmailVerificationBackwardCompatibilityService(
+            app_info,
+            ev_create_and_send_custom_email
+        )
+
+        # Setup EmailPassword (EP) backward compatibility service
+        ep_recipeInterfaceImpl = cast(EPRecipeImplementation, recipe_interface_impl)
+        if email_verification_feature is not None:
+            async def ep_get_email_verification_url(ep_user: EPUser, user_context: Dict[str, Any]):
+                user = User(ep_user.user_id, ep_user.email, ep_user.time_joined)
+                if email_verification_feature.get_email_verification_url is None:
+                    return ""
+                return await email_verification_feature.get_email_verification_url(user, user_context)
```

Comment:
This is irrelevant.

Reply:
Removed. Passing `None`. Is this okay?

Reply:
Nope. See node PR. This file has changed fully.